modules/standard/GMP.chpl

```diff
@@ -560,7 +560,7 @@ module GMP {
     }
     proc set_str(str:string, base:int=0)
     {
-      on this do mpz_set_str(this.mpz, str.c_str(), base.safeCast(c_int));
+      on this do mpz_set_str(this.mpz, str.localize().c_str(), base.safeCast(c_int));
```

Comment:
To clarify, I don't object to the calls, but wanted to make sure I understood the current and proposed behavior (and continue to think that the difference is worth elevating as part of "awareness of what changes the string patch brings" since people writing such code in the future will need to take an action like this.

While I think I agree that c_str() shouldn't automatically localize, I could imagine a helper method that combined localize + c_str() as a helper method, given that it looks likely to be a common pattern, at least for internal devs.  But I'm not suggesting we do that before merging.